### PR TITLE
[TT-12495] Add support for RSASSA-PSS signed JWTs

### PR DESCRIPTION
**Pull request by sedkis**
<details open>
  <summary><a href="https://tyktech.atlassian.net/browse/TT-12495" title="TT-12495" target="_blank">TT-12495</a></summary>
  <br />
  <table>
    <tr>
      <th>Summary</th>
      <td>JWT RSA PUB Improvement - Support RSAPSS</td>
    </tr>
    <tr>
      <th>Type</th>
      <td>
        Story
        Story
      </td>
    </tr>
    <tr>
      <th>Status</th>
      <td>In Code Review</td>
    </tr>
    <tr>
      <th>Points</th>
      <td>N/A</td>
    </tr>
    <tr>
      <th>Labels</th>
      <td><a href="https://tyktech.atlassian.net/issues?jql=project%20%3D%20TT%20AND%20labels%20%3D%20innersource%20ORDER%20BY%20created%20DESC" title="innersource">innersource</a></td>
    </tr>
  </table>
</details>

---

Adding support for the more secure RSA-PSS signed JWTS.

## Description

allows for the use of the RSA-PSS signature algorithm commonly referred to as PS256, PS384, PS512.
The change is invisible to existing RSA Public Keyuse cases.  Simply - by using "RSA Public Key" signing algorithm, Tyk will now validate JWTs signed by both RS & PS Class algorithms using Public Keys.

## Motivation and Context

RSA-PSS is considered more secure than PKCS1 v1.5 due to its probabilistic nature, which helps mitigate certain attacks (e.g., padding oracle attacks).

RS256: Commonly used in legacy systems, JWT (JSON Web Tokens), and many existing protocols where backward compatibility is important.
PS256: Recommended for new applications where higher security is desired. It is becoming more widely adopted in modern security protocols.

## How This Has Been Tested

Unit tests have been added.  Both positive + negative tests that test both RS class JWTs and PS class JWTs.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Refactoring or add test (improvements in base code or adds test coverage to functionality)

## Checklist

- [ ] I ensured that the documentation is up to date
- [ ] I explained why this PR updates go.mod in detail with reasoning why it's required
- [ ] I would like a code coverage CI quality gate exception and have explained why

### CODE REVIEW

**Comment by buger** [comment]
<p>
    Let's make that PR title a 💯 shall we? 💪
    </p>
    <p>
    Your <em>PR title</em> and <em>story title</em> look <strong>slightly different</strong>. Just checking in to know if it was intentional!
    </p>
    <table>
      <tr>
        <th>Story Title</th>
        <td>JWT RSA PUB Improvement - Support RSAPSS</td>
      </tr>
      <tr>
          <th>PR Title</th>
          <td>[TT-12495] Add support for RSASSA-PSS signed JWTs</td>
        </tr>
    </table>
    <p>
      Check out this <a href="https://www.atlassian.com/blog/git/written-unwritten-guide-pull-requests">guide</a> to learn more about PR best-practices.
    </p>

File: gateway/mw_jwt_test.go [hunk 1]
```diff
@@ -1584,6 +1584,119 @@ func TestParseRSAKeyFromJWK(t *testing.T) {
...
+kd3qqGElvW/VDL5AaWTg0nLVkjRo9z+40RQzuVaE8AkAFmxZzow3x+VJYKdjykkJ
+0iT9wCS0DRTXu269V264Vf/3jvredZiKRkgwlL9xNAwxXFg0x/XFw005UWVRIkdg
+cKWTjpBP2dPwVZ4WWC+9aGVd+Gyn1o0CLelf4rEjGoXbAAEgAqeGUxrcIlbjXfbc
+mwIDAQAB
+-----END PUBLIC KEY-----`
+
+	_, err := ParseRSAPublicKey([]byte(sample))
+	if err != nil {
+		t.Error("decoding as default ", err)
+	}
```
Comment (titpetric) [inline gateway/mw_jwt_test.go:1601]:
```suggestion
	assert.NoError(t, err)
```

repeats.

**Review by titpetric — APPROVED** [review]
Generally ok, but can you bring tests a little bit up to date:

1. use testify/assert (mostly the chosen testing DSL)
2. assert on values too, mostly they get swallowed with `_`

Take care of this before merging. LGTM :champagne:

**Pull request merged**